app/game/Main.hs

```diff
@@ -64,10 +67,11 @@ cliParser =
     pausedAtStart <- paused
     autoPlay <- autoplay
     speed <- speedFactor
+    debugOptions <- debug
```

Comment:
I suggest something like this:

```suggestion
    initialDebugOptions <- debug
    let debugOptions = Set.union cheatMode initialDebugOptions
```

Then we can get rid of the lambda + record update in the final `return`.

Reply:
I might have underestimated the powers of ApplicativeDo after last issues. I will try this. 👍

Reply:
Nope, this is not accepted by ApplicativeDo.
```
app/game/Main.hs:62:13: error:
    • No instance for (Monad Parser) arising from a use of ‘return’
```

Reply:
Oh, right, because it's referring to `cheatMode` which is the result of a previous action.  Actually, I now understand why `let debug = Set.union cheatMode ...` is not accepted either---for the same reason.

Reply:
Now I don't even understand why the existing code works.

Reply:
@byorgey I imagine GHC constucts a suitable `a` that puts everything in the right place:

```Haskell
(\c d o -> A (c + d) o) <$> c <*> d <*> o
```